layers/+email/mu4e/packages.el

```diff
@@ -46,6 +47,10 @@
         (add-hook 'mu4e-compose-pre-hook 'mu4e/set-account)
         (add-hook 'message-sent-hook 'mu4e/mail-account-reset)))))
 
+(defun mu4e/init-mu4e-alert ()
+  (use-package mu4e-alert
+    :commands (mu4e-alert-enable-notifications)))
+
```

Comment:
I just used `:commands` to defer loading until this function gets called, but, it is still wrong since I should have added the other functions as well. So I should replace `:commands ...` with `:defer t` since and only use commands for non autoloaded functions if I get this right.

I have a question about the names of the layer variables.  The `mu4e-` prefix is used by mu4e, as a result naming variables with that prefix might at some point conflict with `mu4e`.  Is there any other way to name those variables? Maybe spacemacs-mu4e-xxx? (it gets too long though)

Additionally, I did not find any conventions about the customization of layers. Do we need to define layer variables for customizing the behavior of a package or just give information on how to do this in `dotspacemacs/user-config`?

Thank you for the feedback.